### PR TITLE
Adding the Crustacean Mutation Tree to the game, also adds a mutation from that branch to the Insect and Spider pool.

### PR DESCRIPTION
**Pull request by misora31**
#### Summary
Content "Crustacean Mutation tree"

#### Purpose of change
I have added the Crustacean Mutation tree, after this I will be working on adding more capstones to it and reworking the Cattle tree into the Ruminant tree and adding more mutations to the lizard tree.
I personally thought the game could do with a few more people working on the mutation side of the game and this seemed like an easy way to start. I have far more ambitious plans for the future, but this is a good starting point.

#### Describe the solution
Added the crustacean tag to the Frightening, Chitin line with a branch at CHITIN2 into CRUSTACEAN_CARAPACE instead of CHITIN3, Dense Bones, Heat Dependent, Very Heat Dependent, Smelly 1/2, Eye Stalks with a new rigid eyestalk for crustaceans at the end, Weak Stomach, Nausea, Vomitous, Ugly, Deformed 1/2, Fast Healer, Thirst 1/2, Gills, Clumsy, Eat Poison, Eat Dead, Mut_Tough 1/2, Large and Huge mutation lines, Tolerance,  and Strong to Insanely Strong mutations.

Added the new mutation Ecdysis which allows insects, spiders, and crustaceans to occasionally shed their carapace and added the giant pincer mutation which adds a natural, encumbering weapon that branches off CRUSTACEAN_CARAPCE.

#### Describe alternatives you've considered

Different more anatomically correct mutations done by someone that knows more than I do.

#### Testing

I spent several hours making sure things spawned correctly and that you could correctly mutate through the crustacean branch.

#### Additional context

N/A

### CODE REVIEW

**Comment by fairyarmadillo** [comment]
Exciting! I would recommend changing the claw to be an integrated unarmed weapon though. See this PR for examples: https://github.com/CleverRaven/Cataclysm-DDA/pull/65070

The "occasionally attack" mutations like fangs and hooves are sort of wonky and the devs have signaled that they want to move away from that style and replace them with integrated weapons or parts that give you an extra martial arts style move.

**Comment by misora31** [comment]
> Exciting! I would recommend changing the claw to be an integrated unarmed weapon though. See this PR for examples: #65070
> 
> The "occasionally attack" mutations like fangs and hooves are sort of wonky and the devs have signaled that they want to move away from that style and replace them with integrated weapons or parts that give you an extra martial arts style move.

Will do! I'm new to doing this, so thanks for the advice.

**Comment by Maleclypse** [comment]
In the future if you don't delete the inside [ ] words you can see what you've done wrong and won't need someone else to fix your PR summary for you. :)

**Comment by Maleclypse** [comment]
You are going to want to copy each individual file one by one into https://dev.narc.ro/cataclysm/format.html and then hit the lint button.  If you don't receive an error message you copy the results whether they look identical or not to you back over the original file then save it.  That will fix your linting errors.

**Comment by fairyarmadillo** [comment]
Sorry to hit you with so many suggestions, but you might also want to re-evaluate how you're doing the chitin carapace mutation. Armor mutations have also moved to using integrated items rather than having their armor values in the mutation itself. Your best bet is to copy the way the sclerotin plate works. You can find the item for it in /data/json/items/armor/integrated.json

30 encumbrance is also far too much to be useful. Sclerotin plate is already very powerful and it's only 5-15 encumbrance. I'd suggest copying the values from sclerotin plate and just very slightly increasing its thickness and encumbrance, like maybe .5mm thicker and 2 or 3 more encumbrance in a couple of places to start with. It will make a huge difference even at low values.

**Comment by misora31** [comment]
> Sorry to hit you with so many suggestions, but you might also want to re-evaluate how you're doing the chitin carapace mutation. Armor mutations have also moved to using integrated items rather than having their armor values in the mutation itself. Your best bet is to copy the way the sclerotin plate mutation works. You can find the item entry for it in /data/json/items/armor/integrated.json
> 
> 30 encumbrance is also far too much to be useful on a full-body armor mutation. Sclerotin plate is already very powerful and it's only 5-15 encumbrance. I'd suggest copying the values from sclerotin plate and just very slightly increasing its thickness and encumbrance, like maybe .5mm thicker and 2 or 3 more encumbrance in a couple of places to start with. It will make a huge difference even at low values.
> 
> The high encumbrance on the crab claw mutation is probably fine though, since it's just one part.

I'll take a look at it to change it, and you don't need to apologize I'm always looking for suggestions and ideas to make my work better.

**Comment by fairyarmadillo** [comment]
I've thrown together a mutation for this that might help, if you want to add it:

  "type": "mutation",
  "id": "EYESTALKS_RIGID",
  "name": {
    "str": "Rigid Eyestalks"
  },
  "points": 2,
  "visibility": 8,
  "ugliness": 8,
  "description": "You have compound eyes on rigid stalks that can fold back against your head to protect them from danger. This gives you better all-around vision and allows you to see underwater, but interferes with eyewear. Perception +1",
  "types": [
    "EYES"
  ],
  "armor": [
    {
      "parts": "eyes",
      "cut": 4,
      "bash": 2
    }
  ],
  "restricts_gear": [
    "eyes"
  ],
  "passive_mods": {
    "per_mod": 2
  },
  "flags": [
    "EYE_MEMBRANE"
  ],
  "category": [
    "CRUSTACEAN"
  ]

I would make EYESTALKS1 a prereq for it, like it is on gastropod. This'll let crab people see underwater and also it will be fun to have weird crab eyes.

**Comment by misora31** [comment]
> I've thrown together a mutation for this that might help, if you want to add it:
> 
> "type": "mutation", "id": "EYESTALKS_RIGID", "name": { "str": "Rigid Eyestalks" }, "points": 2, "visibility": 8, "ugliness": 8, "description": "You have compound eyes on rigid stalks that can fold back against your head to protect them from danger. This gives you better all-around vision and allows you to see underwater, but interferes with eyewear. Perception +1", "types": [ "EYES" ], "armor": [ { "parts": "eyes", "cut": 4, "bash": 2 } ], "restricts_gear": [ "eyes" ], "passive_mods": { "per_mod": 1 }, "flags": [ "EYE_MEMBRANE" ], "category": [ "CRUSTACEAN" ]
> 
> I would make EYESTALKS1 a prereq for it. This'll let crab people see underwater and also it will be fun to have weird crab eyes.
> 
> Note that this doesn't use integrated armor, and just slaps a small bonus on. That's partly because the eyes are retractable so they're avoiding damage that way and I think that's probably more appropriate for little things like this, but it would probably work fine the other way too.

Thanks, I'll take a look at it more thoroughly tomorrow. If you'd like a more reliable means of communication I am usually online on discord. I'm in both devcord and the community server as Misora#0001.

File: data/json/items/comestibles/mutagen.json [hunk 1]
```diff
@@ -429,6 +429,21 @@
       "vitamins": [ [ "mutagen_rabbit", 450, 550 ] ]
     }
   },
+  {
+    "id": "iv_mutagen_crustacean",
+    "copy-from": "iv_mutagen_flavor",
+    "type": "COMESTIBLE",
+    "name": { "str": "crustacean mutagenic primer" },
+    "description": " A red colored mutagenic primer.",
```
Comment (Maleclypse) [inline data/json/items/comestibles/mutagen.json:437]: Do primers not tell you what mutagen they contain?

**Comment by bombasticSlacks** [comment]
Your item for the integrated carapace is impossibly heavy, it ways more than a perfect cube of the materials it is made of, you will have to make it less heavy, have more volume, or be made of different materials.

Since it's integrated the values aren't crucial, but it still needs to pass the test since it is an item.

File: data/json/items/armor/integrated.json [hunk 2]
```diff
@@ -707,6 +707,71 @@
     ],
     "melee_damage": { "bash": 4, "cut": 8 }
   },
+  {
+    "id": "integrated_carapace",
+    "type": "ARMOR",
+    "category": "armor",
+    "name": { "str_sp": "sclerotin carapace" },
+    "description": "When the time comes to prey, this unwavering carapace will keep you safe from everything that could harm you.",
+    "weight": "11.4 kg",
```
Comment (Maleclypse) [inline data/json/items/armor/integrated.json:716]:
```suggestion
    "weight": "11 kg",
```

**Pull request merged**